### PR TITLE
fix(schematics): library schematic properly removes extra files

### PR DESCRIPTION
**Pull request by ThomasBurleson**
Angular CLI now supports `host.delete()`. *@nrwl/schematics ngrx* `updateProject()` should properly remove the `*.service` and `*.component` files generated in the library root.

*  @see [`collection/library/index.ts::updateProject()` ](https://github.com/nrwl/nx/pull/658/files#diff-dd0c1903a3c8a563d624a84caa03be0dL228)

>  Earlier workarounds using `unlink()` were not working.

### CODE REVIEW

**Comment by ThomasBurleson** [comment]
@FrozenPandaz - This PR adds a fix to your PR #650. It adds an important file cleanup after library generation.

File: e2e/schematics/command-line.test.ts [hunk 1]
```diff
@@ -175,7 +175,7 @@ describe('Command line', () => {
       );
 
       const output = runCommand(
-        'npm run workspace-schematic -- custom mylib --directory=dir'
+        'npm run workspace-schematic custom mylib -- --directory=dir'
```
Comment (FrozenPandaz) [inline e2e/schematics/command-line.test.ts]: I think some of your other changes snuck in here

File: packages/schematics/src/collection/library/index.ts [hunk 2]
```diff
@@ -225,16 +225,23 @@ function updateNgPackage(options: NormalizedSchema): Rule {
...
-    // host.delete(`${options.projectRoot}/src/lib/${options.name}.component.ts`);
-    // host.delete(`${options.projectRoot}/src/lib/${options.name}.component.spec.ts`);
+    const libRoot = `${options.projectRoot}/src/lib/`;
+
+    host.delete(path.join(libRoot, `${options.name}.service.ts`));
+    host.delete(path.join(libRoot, `${options.name}.service.spec.ts`));
+    host.delete(path.join(libRoot, `${options.name}.component.ts`));
+    host.delete(path.join(libRoot, `${options.name}.component.spec.ts`));
+
+    if (!publishable) {
```
Comment (FrozenPandaz) [inline packages/schematics/src/collection/library/index.ts]: We can read publishable from options:
```ts
if (!options.publishable) {
```

File: packages/schematics/src/collection/library/index.ts [hunk 3]
```diff
@@ -370,56 +377,14 @@ export default function(schema: Schema): Rule {
       throw new Error(`routing must be set`);
     }
 
-    // @angular-devkit/core doesn't allow us to delete files
-    setTimeout(() => {
```
Comment (FrozenPandaz) [inline packages/schematics/src/collection/library/index.ts]: Woohoo

**Review by FrozenPandaz — COMMENTED** [review]
Nice! This works well!

File: e2e/schematics/command-line.test.ts (hunk 1 above)
Comment (vsavkin) [reply e2e/schematics/command-line.test.ts]: @ThomasBurleson do you know how this works? I thought everything after -- gets passed to your script, and everything before -- gets consumed by npm itself.

File: packages/schematics/src/collection/library/index.ts (hunk 2 above)
Comment (vsavkin) [reply packages/schematics/src/collection/library/index.ts]: NormalizedSchema extends Schema, which has publishable

File: packages/schematics/src/collection/library/index.ts [hunk 4]
```diff
@@ -370,56 +377,14 @@ export default function(schema: Schema): Rule {
...
-
     return chain([
       externalSchematic('@schematics/angular', 'library', {
         name: options.name,
         prefix: options.prefix,
         entryFile: 'index'
       }),
       move(options.name, options.projectRoot),
-      updateProject(options),
+      updateProject(options, schema.publishable),
```
Comment (vsavkin) [inline packages/schematics/src/collection/library/index.ts]: remove schema.publishable

**Review by vsavkin — CHANGES_REQUESTED** [review]
Looks good. One minor change.

**Comment by vsavkin** [comment]
Merged here: 8334c200a79575a25b711ce6e58164a331295a35

**Pull request closed without merging**